### PR TITLE
Finalizer for MSession and fix windows persistent engine handling

### PR DESCRIPTION
**Pull request by musm**
Fix https://github.com/JuliaInterop/MATLAB.jl/issues/65

### CODE REVIEW

**Comment by musm** [comment]
Please review and merge after https://github.com/JuliaInterop/MATLAB.jl/pull/60

File: src/MATLAB.jl [hunk 1]
```diff
@@ -46,11 +45,14 @@ include("engine.jl")
...
 
 function __init__()
+    if is_windows()
+    global persistent_msession
     # workaround "primary message table for module 77" error
     # creates a dummy Engine session and keeps it open so the libraries used by all other
     # Engine clients are not loaded and unloaded repeatedly
     # see: https://www.mathworks.com/matlabcentral/answers/305877-what-is-the-primary-message-table-for-module-77
-    is_windows() && MSession(0)
+    persistent_msession = MSession(0)
```
Comment (musm) [inline src/MATLAB.jl]: The user should never touch this/delete this reference.

File: src/engine.jl [hunk 2]
```diff
@@ -56,7 +73,7 @@ default_msession = nothing
 
 function restart_default_msession(bufsize::Integer = default_output_buffer_size)
     global default_msession
-    if default_msession !== nothing
+    if default_msession !== nothing && default_msession.ptr != C_NULL
```
Comment (musm) [inline src/engine.jl]: perhaps there is a better way but we must also check that the `default_msession` has not been `close` but the user resulting in a `C_NULL` ptr

File: src/MATLAB.jl [hunk 3]
```diff
@@ -45,11 +45,14 @@ include("engine.jl")
 include("matstr.jl")
 
 function __init__()
+    if is_windows()
+    global persistent_msession
```
Comment (yuyichao) [inline src/MATLAB.jl]: Indent is off here.

File: test/engine.jl [hunk 4]
```diff
@@ -52,3 +52,9 @@ delete(x)
 @test_throws UndefRefError is_numeric(x)
 @test_throws UndefRefError jscalar(x)
 @test_throws UndefRefError jvariable(x)
+
+# make sure restart_default_msession() doesn't error on null references on
+# default msession
+s = get_default_msession()
+close(s)
+restart_default_msession()
```
Comment (yuyichao) [inline test/engine.jl]: And missing trailing new line...

**Pull request merged**